src/main/java/org/example/dao/PointDAOImpl.java

```diff
@@ -0,0 +1,104 @@
+package org.example.dao;
+
+import org.example.configurations.ConnectionPool;
+import org.example.model.Point;
+
+import java.sql.*;
+import java.util.ArrayList;
+import java.util.List;
+
+
+public class PointDAOImpl implements PointDAO {
+    private static final ConnectionPool CONNECTIONPOOL = ConnectionPool.getInstance();
+
+    public PointDAOImpl() {
+    }
+
+    @Override
+    public void insertPoint(Point point) {
+        Connection connection = CONNECTIONPOOL.getConnectionFromPool();
+        try {
+            PreparedStatement statement = connection.prepareStatement("INSERT INTO point (id, xCoordinate, yCoordinate) VALUES (?, ?, ?)", Statement.RETURN_GENERATED_KEYS);
+            statement.setInt(1, point.getId());
+            statement.setDouble(2, point.getXCoordinate());
+            statement.setDouble(3, point.getYCoordinate());
+            statement.executeUpdate();
+
+            ResultSet generatedKeys = statement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                int id = generatedKeys.getInt(1);
+                point.setId(id);
+                System.out.println("Point inserted successfully. Generated ID: " + id);
+            }
+
+            statement.close();
+            generatedKeys.close();
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }finally {
+            CONNECTIONPOOL.releaseConnectionToPool(connection);
+        }
+    }
+
+    @Override
+    public void deletePoint(Point point) {
```

Comment:
Maybe let's make it delete by id to make it simpler and more unified since we're doing get by id below?

Reply:
Agreed

Reply:
done